x-pack/plugins/reporting/server/config/create_config.ts

```diff
@@ -33,14 +33,18 @@ export function createConfig(
     encryptionKey = crypto.randomBytes(16).toString('hex');
   }
 
-  const { kibanaServer: reportingServer } = config;
+  const { kibanaServer: reportingServer, statefulSettings } = config;
   const serverInfo = core.http.getServerInfo();
   // set kibanaServer.hostname, default to server.host, don't allow "0.0.0.0" as it breaks in Windows
   let kibanaServerHostname = reportingServer.hostname
     ? reportingServer.hostname
     : serverInfo.hostname;
 
-  if (
+  if (statefulSettings.enabled === false) {
+    logger.warn(
+      `Found 'server.host: "0.0.0.0"' in Kibana configuration. Overriding 'xpack.reporting.kibanaServer.hostname' with 'localhost'`
+    );
+  } else if (
```

Comment:
We can "soften" the log message and take out the mention of internal implementation details. This can be a general change, not just for serverless.
```typescript
  if (
    ipaddr.isValid(kibanaServerHostname) &&
    !sum(ipaddr.parse(kibanaServerHostname).toByteArray())
  ) {
    // user can add 'xpack.reporting.kibanaServer.hostname: localhost' in kibana.yml to prevent this log
    logger.info(
      `Overriding server host address "0.0.0.0" in Reporting runtime config,` +
        ` using "xpack.reporting.kibanaServer.hostname: localhost".`
    );
    kibanaServerHostname = 'localhost';
  }
```

I think it would be better to use `info` rather than `warn` because we don't need to portray it as "something went wrong." It's just a corner case dealing with config, that has a totally acceptable code workaround.

Reply:
Also, I wonder why the `kibanaServer` config is under `reporting` and not under `screenshotting`. I think it was a forgotten part of moving the screenshotting code out of reporting. The kibanaServer config data is only needed for the headless chromium browser controlled by the screenshotting plugin, so moving this over there would be a good thing to do at some point.